addons/l10n_multilang/models/l10n_multilang.py

```diff
@@ -87,11 +87,20 @@ def _process_fiscal_pos_translations(self, company_id, langs, field):
         return self.process_translations(langs, field, in_ids, out_ids)
 
     def _get_template_from_model(self, company_id, model):
+        """ Find the records and their matching template """
+        # genereated records have an external id with the format <company id>_<template id>
         out_data = self.env['ir.model.data'].search([('model', '=', model), ('name', '=like', str(company_id)+'\_%')])
```

Comment:
Shouldn't we be using the module name qualifier when searching in `ir.model.data`? The unique constraint only guarantees that the `(module, name)` pair is unique, so there is a non-negligible chance of collision in the names, especially as we're talking of the same kinds of records in different CoA. The generated records share the same module name as their template, AFAIK.

Reply:
The problem is that we are not working with one module but globally (i.e. after the activation of a language), we don't have the module information. We are supposed to have only one CoA per company so the name conflicts should not happen, right?

Reply:
> We are supposed to have only one CoA per company so the name conflicts should not happen, right?

Sure, one CoA per company. But the IMD table is global, so the second IMD search can possibly find multiple matches for a single name. 
Hypothetical example: when you install a CoA for a second company you end up with an `account.account.type` record with XML ID `l10n_generic_coa.2_a_salary_expense` and thus search for `name='a_salary_expense'` in IMD, and get two results, one for `l10n_generic_coa.a_salary_expense` and another one for `l10n_be.a_salary_expense`. 
But actually the record is fully-qualified, so if you searched within the module namespace you would only get a single match. To fix this I imagine we could simply group the results of the first search by module, and then process them in per-module batches - in most cases there will only be one module.